modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/test/metal/js/Validation/Validation.es.js

```diff
@@ -0,0 +1,340 @@
+import Validation from 'source/Validation/Validation.es';
+import {dom as MetalTestUtil} from 'metal-dom';
+
+let component;
+const spritemap = 'icons.svg';
+
+describe(
+	'Validation',
+	() => {
+		afterEach(
+			() => {
+				if (component) {
+					component.dispose();
+				}
+			}
+		);
+
+		it(
+			'should render checkbox to enable Validation',
+			() => {
+				component = new Validation(
+					{
+						dataType: 'string',
+						label: 'Validator',
+						name: 'validation',
+						spritemap
+					}
+				);
+
+				expect(component).toMatchSnapshot();
+			}
+		);
+
+		it(
+			'should enable validation after click on toogle',
+			() => {
+				jest.useFakeTimers();
+
+				component = new Validation(
+					{
+						dataType: 'string',
+						label: 'Validator',
+						name: 'validation',
+						spritemap
+					}
+				);
+
+				MetalTestUtil.triggerEvent(
+					component.element.querySelector('input[type=checkbox]'),
+					'change',
+					{}
+				);
+
+				jest.runAllTimers();
+
+				expect(component).toMatchSnapshot();
+			}
+		);
+
+		it(
+			'should render parameter field with TextField element',
+			() => {
+				jest.useFakeTimers();
+
+				component = new Validation(
+					{
+						dataType: 'string',
+						enableValidation: true,
+						label: 'Validator',
+						name: 'validation',
+						spritemap
+					}
+				);
+
+				component.handleValidationValue({value: ['notContains']});
+
+				jest.runAllTimers();
+
+				expect(component).toMatchSnapshot();
+			}
+		);
+
+		it(
+			'should render parameter field with Numeric element',
+			() => {
+				jest.useFakeTimers();
+
+				component = new Validation(
+					{
+						dataType: 'numeric',
+						enableValidation: true,
+						label: 'Validator',
+						name: 'validation',
+						spritemap
+					}
+				);
+
+				component.handleValidationValue({value: ['lt']});
+
+				jest.runAllTimers();
+
+				expect(component).toMatchSnapshot();
+			}
+		);
+
+		it(
+			'should render parameter field with Text element and then with Numeric after update dataType',
+			() => {
+				jest.useFakeTimers();
+
+				component = new Validation(
+					{
+						dataType: 'string',
+						enableValidation: true,
+						label: 'Validator',
+						name: 'validation',
+						spritemap,
+						validation: {
+							dataType: undefined
```

Comment:
maybe you can just parse `validation: {}` on the Validation component props instead of using: 

```
validation: {
   dataType: undefined
}